escrow/SPECIFICATION.md

```diff
@@ -0,0 +1,95 @@
+Table of Content
+- [Actions that users are able to perform](#actions-that-users-are-able-to-perform)
+  - [Arbiter](#arbiter)
+    - [`resolve_dispute()`](#resolve_dispute)
+  - [Buyer](#buyer)
+    - [`accept_arbiter()`](#accept_arbiter)
+    - [`deposit()`](#deposit)
+    - [`dispute()`](#dispute)
+    - [`transfer_to_seller()`](#transfer_to_seller)
+  - [Seller](#seller)
+    - [`create_escrow()`](#create_escrow)
+    - [`propose_arbiter()`](#propose_arbiter)
+    - [`return_deposit()`](#return_deposit)
+    - [`take_payment()`](#take_payment)
+
+# Overview
+
+This document provides an overview of the application.
+
+It outline the use cases, i.e. desirable functionality, in addition to requirements for the smart contract and the user interface.
+
+# Use Cases
+
+This section contains general information about the functionality of the application and thus does not touch upon any technical aspects.
+
+If you are interested in a functional overview then this is the section for you.
+
+## Actions that users are able to perform
+
+This sub-section details what a user is able to do e.g. click a button and "x, y, z" happens.
+
+### Arbiter
+
+#### `resolve_dispute()`
+
+1. When an escrow is in dispute the arbiter of the escrow can perform a resolution in either direction of the buyer or seller
+2. A resolution consists of
+   1. Selecting either the buyer or seller to whom the deposited asset from the buyer will be sent to
+   2. Selecting an amount to take as payment from the collateral provided by the seller
+
+### Buyer
+
+#### `accept_arbiter()`
+
+1. The buyer is able to accept a proposition by the seller to change the arbiter or change the fee for the arbiter
+
+> **NOTE** They are not required to do so however there may be instances where changing the arbiter / fee is favourable
+
+#### `deposit()`
+
+1. The buyer must deposit into the escrow one asset from the options provided by the seller
+2. The escrow accepts a 1 time deposit therefore the buyer cannot go back and deposit a different asset
+
+> **NOTE** Once a buyer deposits they are unable to get the asset out by themselves. Either the arbiter must resolve in their favour or the seller must return the deposit. This is a safety mechanism.
+
+#### `dispute()`
+
+1. The buyer is able to `dispute()` the escrow which prevents the seller from taking their deposit after the deadline
+2. Disputing means that there are 3 ways to move the deposit from the buyer out of the escrow
+   1. The buyer transfers to seller
+   2. The seller returns the deposit to the buyer
+   3. The arbiter may come in and resolve the dispute in favour of the buyer or seller
```

Comment:
It may be useful to a reader to include the functions that perform the operations mentions (`transfer_to_seller`) etc

Reply:
> to include the functions that perform the operations mentions

I do not understand.

I am planning on adding diagrams which show a visual contract flow of all the use cases

Reply:
I mean it may be useful to mention the functions that perform the 3 different ways to move the deposit from the buyer out of the escrow.  `transfer_to_seller`, `resolve_dispute`, and `return_deposit`.  Although if you add diagrams in the future this may not be a useful addition.